elasticsearch/src/main/java/org/hibernate/search/elasticsearch/schema/impl/model/IndexMetadata.java

```diff
@@ -20,17 +21,17 @@
  */
 public class IndexMetadata {
 
-	private String name;
+	private URLEncodedString name;
```

Comment:
It would be a bit clearer if we used a dedicated `IndexIdentifier` type, able to return a simple string or an encoded one. Mainly because it's really not obvious why the name is URL-encoded, both here and in the IndexManager.

... but it would once again require two hours of find and replace, so I think we'll stick with this :p